### ISSUE TITLE
Is it possible to implement multi-agent traffic single with DQN?

### ISSUE THREAD

**Issue by ZhongmingGit** [LucasAlegre/sumo-rl#2]
Hi, I saw the experiment dqn_2way-single-intersection.py, which implement by stable_baselines.deepq.  Is it possible to implement multi-agent traffic single control in a 2x2 grid map by it? Sorry, I'm not familiar with stable_baselines. 

And the current code sets the same signal phase in a 2x2 grid map, have you considered that signal lights in one 2x2 grid map may be different?

Thanks
Ming

**Comment by LucasAlegre** [comment]
Hi,

Unfortunately stable-baselines doesn't provide support for multiagent RL :(

However, rllib has great support: https://bair.berkeley.edu/blog/2018/12/12/rllib/
I used their version of A3C here: https://github.com/LucasAlegre/sumo-rl/blob/master/experiments/a3c_4x4grid.py
It is possible that all traffic light agents follow the same policy (but still act decentralized, as each traffic light has its own observation) or that each agent has its own policy.

Lucas

**Comment by ZhongmingGit** [comment]
Thank you very much for your help

I have tested all A3C version,  all code came the same error:

Step #120.00Error: tcpip::Socket::recvAndCheck @ recv: peer shutdownUF 24)    
Quitting (on error).

My sumo version is 1.1.0. I have tried to turn off the firewall and other methods, but useless. I think this may be caused by the repeated occupancy of the port.
Have you met this problem before?

Thanks
Ming
